Library/Homebrew/formula.rb

```diff
@@ -1835,6 +1832,23 @@ def eligible_kegs_for_cleanup
 
   private
 
+  def declared_runtime_dependencies
+    recursive_dependencies do |_, dependency|
+      Dependency.prune if dependency.build?
+      Dependency.prune if !dependency.required? && build.without?(dependency)
+    end
+  end
+
+  def undeclared_runtime_dependencies
+    return [] unless optlinked?
+
+    keg = Keg.new(opt_prefix)
+    linkage_checker = LinkageChecker.new(keg, self)
+    dylib_formula_names = linkage_checker.brewed_dylibs.keys
```

Comment:
This includes the declared ones.

Reply:
Ah, so it does. This didn't matter when I had this all in one method because we took the union, but it does now that it's in a method called `undeclared_runtime_dependencies`.